python/feathub/feature_views/feature_view.py

```diff
@@ -133,25 +133,28 @@ def get_output_fields(self, source_fields: List[str]) -> List[str]:
             raise FeathubException(
                 "Build this feature view before getting output fields."
             )
-        output_fields = []
+
+        derived_fields = [feature.name for feature in self.get_resolved_features()]
+        key_fields = [
+            key
+            for feature in self.get_resolved_features()
+            if feature.keys
+            for key in feature.keys
+        ]
```

Comment:
If there are multiple features, `key_fields` might contain duplicate entries.

Besides, given that derived feature view can have features with different keys, it might be unnecessary to consider key fields when ordering fields.

Reply:
The following `OrderedDict.fromkeys()` would handle the duplicate entries.